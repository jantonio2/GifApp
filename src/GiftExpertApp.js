import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory'
import { GifGrid } from './components/GifGrid';

export const GiftExpertApp = ({defaultCategories = []}) => {

  // const [categories, setCategories] = useState(['Game of Thrones']);
  const [categories, setCategories] = useState(defaultCategories);
  
  // const handleAdd = () => {
  //   // setCategories([...categories, 'Game of Thrones']);
  //   setCategories(categs => [...categs, 'Game of Thrones']);
  // };
  
  return (
    <>
      <h2>GiftExpertApp</h2>
      
      <AddCategory setCategories = {setCategories} />

      <hr />

      <ol>
        {
          categories.map(category => 
            <GifGrid 
              key = {category}
              category = {category} 
            />
          )
        }
      </ol>
    </>
  )
}
