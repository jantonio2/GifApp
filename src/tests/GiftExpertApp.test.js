import { shallow } from 'enzyme';
import { GiftExpertApp } from '../GiftExpertApp';

describe('Probando componente GiftExpertApp', () => {
  test('debe de renderizar y hacer match con el snapshot', () => {
    const wrapper = shallow(<GiftExpertApp />);

    expect(wrapper).toMatchSnapshot();
  });

  test('debe de mostrar una lista de categorias', () => {
    const categories = ['Game of Thrones', 'Dragon Ball'];
    const wrapper = shallow(<GiftExpertApp defaultCategories = {categories}/>);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });
  
  
});
