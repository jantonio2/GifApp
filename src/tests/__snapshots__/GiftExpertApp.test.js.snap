// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando componente GiftExpertApp debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Game of Thrones"
      key="Game of Thrones"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`Probando componente GiftExpertApp debe de renderizar y hacer match con el snapshot 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
